runtime/rococo/src/governance/fellowship.rs

```diff
@@ -0,0 +1,355 @@
+				},
+			),
+			(
+				9u16,
+				pallet_referenda::TrackInfo {
+					name: "grand masters",
+					max_deciding: 10,
+					decision_deposit: 1 * 3 * CENTS,
+					prepare_period: 30 * MINUTES,
+					decision_period: 7 * DAYS,
+					confirm_period: 30 * MINUTES,
+					min_enactment_period: 1 * MINUTES,
+					min_approval: pallet_referenda::Curve::LinearDecreasing {
+						length: Perbill::from_percent(100),
+						floor: Perbill::from_percent(50),
+						ceil: Perbill::from_percent(100),
+					},
+					min_support: pallet_referenda::Curve::LinearDecreasing {
+						length: Perbill::from_percent(100),
+						floor: Perbill::from_percent(0),
+						ceil: Perbill::from_percent(50),
+					},
+				},
+			),
+		];
+		&DATA[..]
+	}
+	fn track_for(id: &Self::RuntimeOrigin) -> Result<Self::Id, ()> {
+		use super::origins::Origin;
+
+		#[cfg(feature = "runtime-benchmarks")]
+		{
+			// For benchmarks, we enable a root origin.
+			// It is important that this is not available in production!
+			let root: Self::RuntimeOrigin = frame_system::RawOrigin::Root.into();
+			if &root == id {
+				return Ok(9)
+			}
+		}
+
+		match Origin::try_from(id.clone()) {
+			Ok(Origin::FellowshipInitiates) => Ok(0),
+			Ok(Origin::Fellowship1Dan) => Ok(1),
+			Ok(Origin::Fellowship2Dan) => Ok(2),
+			Ok(Origin::Fellowship3Dan) | Ok(Origin::Fellows) => Ok(3),
+			Ok(Origin::Fellowship4Dan) => Ok(4),
+			Ok(Origin::Fellowship5Dan) | Ok(Origin::FellowshipExperts) => Ok(5),
+			Ok(Origin::Fellowship6Dan) => Ok(6),
+			Ok(Origin::Fellowship7Dan | Origin::FellowshipMasters) => Ok(7),
+			Ok(Origin::Fellowship8Dan) => Ok(8),
+			Ok(Origin::Fellowship9Dan) => Ok(9),
+			_ => Err(()),
+		}
+	}
+}
+pallet_referenda::impl_tracksinfo_get!(TracksInfo, Balance, BlockNumber);
+
+pub type FellowshipReferendaInstance = pallet_referenda::Instance2;
+
+impl pallet_referenda::Config<FellowshipReferendaInstance> for Runtime {
+	type WeightInfo = weights::pallet_referenda_fellowship_referenda::WeightInfo<Self>;
+	type RuntimeCall = RuntimeCall;
+	type RuntimeEvent = RuntimeEvent;
+	type Scheduler = Scheduler;
+	type Currency = Balances;
+	type SubmitOrigin =
+		pallet_ranked_collective::EnsureMember<Runtime, FellowshipCollectiveInstance, 1>;
+	type CancelOrigin = FellowshipExperts;
+	type KillOrigin = FellowshipMasters;
+	type Slash = Treasury;
+	type Votes = pallet_ranked_collective::Votes;
+	type Tally = pallet_ranked_collective::TallyOf<Runtime, FellowshipCollectiveInstance>;
+	type SubmissionDeposit = SubmissionDeposit;
+	type MaxQueued = ConstU32<100>;
+	type UndecidingTimeout = UndecidingTimeout;
+	type AlarmInterval = AlarmInterval;
+	type Tracks = TracksInfo;
+	type Preimages = Preimage;
+}
+
+pub type FellowshipCollectiveInstance = pallet_ranked_collective::Instance1;
+
+morph_types! {
+	/// A `TryMorph` implementation to reduce a scalar by a particular amount, checking for
+	/// underflow.
+	pub type CheckedReduceBy<N: TypedGet>: TryMorph = |r: N::Type| -> Result<N::Type, ()> {
```

Comment:
there is the same type right now in substrate, you can reuse it

Reply:
addressing this at [9e2dbe4](https://github.com/paritytech/polkadot/pull/7272/commits/9e2dbe48052b8ebe170c62f155f15fb54d38cc5a)